src/der.rs

```diff
@@ -312,9 +313,9 @@ pub(crate) fn nested_of_mut<'a>(
     error: Error,
     mut decoder: impl FnMut(&mut untrusted::Reader<'a>) -> Result<(), Error>,
 ) -> Result<(), Error> {
-    nested(input, outer_tag, error, |outer| {
```

Comment:
a bit of a bummer to see the `clone()`s growing in this code path but I think it's unlikely to have a significant perf impact (hopefully benchmarks agree). In either case the benefit of more informative errors seems worth a small sacrifice.